Nuclei SDK passive mode does not work

<!-- 
1. Please search to see if an issue already exists for the bug you encountered.
2. For support requests, FAQs or "How to" questions, please use the GitHub Discussions section instead - https://github.com/projectdiscovery/nuclei/discussions or
3. Join our discord server at https://discord.gg/projectdiscovery and post the question on the #nuclei channel.
-->

<!-- ISSUES MISSING IMPORTANT INFORMATION MAY BE CLOSED WITHOUT INVESTIGATION. -->

### Nuclei version:
<!-- You can find current version of nuclei with "nuclei -version" -->
<!-- We only accept issues that are reproducible on the latest version of nuclei. -->
<!-- You can find the latest version of project at https://github.com/projectdiscovery/nuclei/releases/ -->

v.3.2.9

### Current Behavior:
<!-- A concise description of what you're experiencing. -->

The `nuclei` SDK is not currently able to configure an engine and/or load targets to run passive scans on save HTTP responses.

The current `EnablePassive` configuration option does not appear to be sufficient to make an engine run passive scans.

### Expected Behavior:
<!-- A concise description of what you expected to happen. -->

The SDK API and documentation should provide a way to run a passive scan against stored HTTP responses.

### Steps To Reproduce:
<!--
Example: steps to reproduce the behavior:
1. Run 'nuclei -t ... -u ..'
2. See error...
-->

1. Confirm that passive `nuclei` passive mode works from the cli
```shell
$ httpx-toolkit -sr -u https://juice-shop.herokuapp.com

    __    __  __       _  __
   / /_  / /_/ /_____ | |/ /
  / __ \/ __/ __/ __ \|   /
 / / / / /_/ /_/ /_/ /   |
/_/ /_/\__/\__/ .___/_/|_|
             /_/

		projectdiscovery.io

[INF] Current httpx version v1.6.4 (outdated)
https://juice-shop.herokuapp.com


$ nuclei -passive -target output/response/juice-shop.herokuapp.com/

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v3.2.9

		projectdiscovery.io

[INF] Current nuclei version: v3.2.9 (latest)
[INF] Current nuclei-templates version: v9.9.1 (latest)
[WRN] Scan results upload to cloud is disabled.
[INF] New templates added in latest release: 75
[INF] Templates loaded for current scan: 1171
[INF] Executing 1171 signed templates from projectdiscovery/nuclei-templates
[INF] Targets loaded for current scan: 1
[addeventlistener-detect] [] [info] output/response/juice-shop.herokuapp.com/47a267551a6bb823709fdfb9cf2092d8e282e7ba.txt
[x-recruiting-header] [] [info] output/response/juice-shop.herokuapp.com/47a267551a6bb823709fdfb9cf2092d8e282e7ba.txt
[cowboy-detect] [] [info] output/response/juice-shop.herokuapp.com/47a267551a6bb823709fdfb9cf2092d8e282e7ba.txt
[fingerprinthub-web-fingerprints:qm-system] [] [info] output/response/juice-shop.herokuapp.com/47a267551a6bb823709fdfb9cf2092d8e282e7ba.txt
[owasp-juice-shop-detect] [] [info] output/response/juice-shop.herokuapp.com/47a267551a6bb823709fdfb9cf2092d8e282e7ba.txt
```

2. Try and run the same scan using the SDK

```golang
package main

import (
	"fmt"

	nuclei "github.com/projectdiscovery/nuclei/v3/lib"
	output "github.com/projectdiscovery/nuclei/v3/pkg/output"
)

func handleResultEvent(event *output.ResultEvent) {
	fmt.Printf("%#v/n", event)
}

func main() {
	fmt.Println("Starting new engine")

	ne, err := nuclei.NewNucleiEngine(
		nuclei.EnablePassiveMode(),
	)
	if err != nil {
		panic(err)
	}
	defer ne.Close()

	// This should probably be done in EnablePassiveMode
	ne.Options().DisableHTTPProbe = true

	ne.LoadTargetsWithHttpData(
		"./output/response/localhost_3000/d167406719b748a6d5d052538558eed53735a389.txt",
		"",
	)
	err = ne.ExecuteWithCallback(handleResultEvent)
	if err != nil {
		panic(err)
	}
}
```

```shell
$ go run main.go 
Starting new engine
panic: [:RUNTIME] No targets available

goroutine 1 [running]:
main.main()
	/tmp/tmp.ruOdyy29rH/main.go:35 +0x153
exit status 2
```





### Anything else:
<!-- Links? References? Screenshots? Anything that will give us more context about the issue that you are encountering! -->

- It should be noted the `LoadTargetsWithHttpData` defaults to a `ListInputProvider` when provided a `.txt.` file, and there is currently no such thing as a `TxtResponseInputProvider` (which may be a missing object).   I was not able to prevent the `ListInputProvider` from spltting the http response data by lines.

- I also attempted to set the executor `ProtocolType = OfflineHTTPProtocol` (both on executer and engine options), but this did not work.  It may be that this has to be set before the engine is created.

--- a/lib/config.go
+++ b/lib/config.go
@@ -396,6 +396,7 @@ func WithHeaders(headers []string) NucleiSDKOptions {
 func EnablePassiveMode() NucleiSDKOptions {
 	return func(e *NucleiEngine) error {
 		e.opts.OfflineHTTP = true
+		e.opts.DisableHTTPProbe = true
 		return nil
 	}
 }

